packages/f-checkout/src/components/Selector.vue

```diff
@@ -48,85 +27,34 @@ export default {
             return this.serviceType === CHECKOUT_METHOD_DELIVERY
                 ? this.$t('labels.deliveryOrderMethod')
                 : this.$t('labels.collectionOrderMethod');
-        }
-    },
-
-    watch: {
-        'fulfillment.times' (newValue) {
-            const selected = newValue.find(t => t.selected);
-            if (selected) {
-                this.selectedTime = selected.from;
-            }
+        },
+
+        fulfillmentTimes () {
+            /*
+            * Create an array from fulfillment times labels to
+            * display as options in dropdown
+            */
+            return this.fulfillment.times.map(time => time.label.text);
         }
     },
 
     methods: {
-        selectionChanged () {
-            this.fulfillment.times.forEach(el => { el.selected = false; });
-            const newSelected = this.fulfillment.times.find(t => t.from === this.selectedTime);
-            if (newSelected) {
-                newSelected.selected = true;
-            }
+        selectionChanged (e) {
+            /*
+            * Update all fulfillment.times.selected to false
+            * Update chosen fulfillment.times.selected to true
+            */
+            this.fulfillment.times.forEach(time => {
+                time.selected = time.label.text === e;
```

Comment:
Is this line correct? Seems to be comparing the `time.label.text` to the event object that Vue returns?

Reply:
@kevinrodrigues Yeah, it's testing that if it finds a time that matches the event object it will change the fulfillment.times object so that that time has a selected value of true then changing the rest to false.
https://github.com/justeat/fozzie-components/pull/473#discussion_r533218552

Reply:
Ah I've just realised why it doesn't contain the preventDefault stuff. It comes from an emitted event from `f-form-field` that just sends the time. `e` is probably not the best name for the variable. I'll change it

Reply:
Yeah that now makes sense, sorry partially me not knowing the the code fully. I think if we name it something else it would make it easier to follow. I'd also add a param to the jsDoc above so we know what value we're expecting here e.g `@param {time}` :)

Reply:
I've updated the description now too. Does it make more sense like that?

Reply:
Yup looks good now 👍